src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -78,6 +75,23 @@ public HttpMethod(string method)
             _method = method;
         }
 
+        // Private constructor used internally to create an HttpMethod with a pre-calculated
+        // hash code. The method parameter must already be uppercase, which allows us to use
+        // method.GetHashCode() instead of method.ToUpperInvariant().GetHashCode(), avoiding
+        // the unnecessary extra string allocation. The ignored parameter is just there to
+        // distinguish this constructor overload from the public constructor.
+        private HttpMethod(string method, bool ignored)
```

Comment:
@justinvp To avoid duplication and risk of getting the two ctors out of sync, we can move the code within the private one then chain them using the last bool param: if the param is set to true then _hashcode will be initialized.